dvc/path_info.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import unicode_literals
+
+import sys
+import os
+import posixpath
+from funcy import cached_property
+
+from dvc.utils.compat import str, basestring, is_py2
+from dvc.utils.compat import pathlib, fspath_py35, urlparse
+
+
+if is_py2:
+    fs_encoding = sys.getfilesystemencoding()
+
+
+class PathInfo(pathlib.PurePath):
+    scheme = "local"
+
+    def __new__(cls, *args):
+        # Construct a proper subclass depending on current os
+        if cls is PathInfo:
+            cls = WindowsPathInfo if os.name == "nt" else PosixPathInfo
+        return cls._from_parts(args)
+
+    @classmethod
+    def from_posix(cls, s):
+        return cls(pathlib.PurePosixPath(s))
+
+    def __str__(self):
+        path = self.__fspath__()
+        if os.name == "nt" and not os.path.commonprefix([os.getcwd(), path]):
+            # In case of windows, when cache is on different drive than
```

Comment:
can we describe not in some DVC related terms (like cache), this is "library" class, takes a while to actually understand what cache has to do, what's the problem actually is.

Reply:
we can keep cache as an example when it becomes important